core/common/src/serializers/DateTimePeriodSerializers.kt

```diff
@@ -95,7 +95,7 @@ public object DateTimePeriodIso8601Serializer: KSerializer<DateTimePeriod> {
 /**
  * A serializer for [DatePeriod] that uses a different field for each component, only listing non-zero components.
  *
- * Deserializes the time components as well, as long as they are zero.
+ * Deserializes the time components as well ensuring they are zero.
```

Comment:
I think this obfuscates the fact that the time components are optional. Instead, it looks like the serializer makes sure that they are present and are zero.